template/build/core/phing/tasks/deploy.xml

```diff
@@ -31,29 +43,29 @@
     <phingcall target="deploy:artifact:push-all"/>
   </target>
 
+  <target name="deploy:artifact:add-remotes" description="Add remotes and fetch upstream refs.">
+      <foreach list="${deploy.remote.gits}" param="deploy.remote.git" target="deploy:artifact:add-remote"/>
+  </target>
+
   <target name="deploy:artifact:add-remote" description="Adds a git remote and checks out deploy branch from upstream.">
-    <echo message="Fetching from git remote ${deploy.remote}"/>
+    <echo message="Fetching from git remote ${deploy.remote.git}"/>
 
     <!-- Generate and md5 sum of the remote URL to use as remote name. -->
-    <exec command="echo ${deploy.remote} | openssl md5 | cut -d' ' -f 2" outputProperty="remoteName"/>
-    <exec command="git remote add ${remoteName} ${deploy.remote}" dir="${deploy.dir}" logoutput="true" passthru="true"/>
-    <exec command="git fetch ${remoteName}" dir="${deploy.dir}" logoutput="true" passthru="true"/>
+    <exec command="echo ${deploy.remote.git} | openssl md5 | cut -d' ' -f 2" outputProperty="deploy.remote.hash"/>
+    <exec command="git remote add ${deploy.remote.hash} ${deploy.remote.git}" dir="${deploy.dir}" logoutput="true" passthru="true"/>
+    <exec command="git fetch ${deploy.remote.hash}" dir="${deploy.dir}" logoutput="true" passthru="true"/>
 
     <!-- Create the new branch. -->
     <exec command="git checkout -b ${deploy.branch}" dir="${deploy.dir}" logoutput="true" passthru="true"/>
     <!-- Pull the latest updates (if available). -->
-    <exec command="git merge ${remoteName}/${deploy.branch}" dir="${deploy.dir}" logoutput="true" passthru="true"/>
-  </target>
-
-  <target name="deploy:artifact:add-remotes" description="Add remotes and fetch upstream refs.">
-    <foreach list="${git.remotes}" param="deploy.remote" target="deploy:artifact:add-remote"/>
+    <exec command="git merge ${deploy.remote.hash}/${deploy.branch}" dir="${deploy.dir}" logoutput="true" passthru="true"/>
```

Comment:
I think you still want to only do this once rather than merge from every remote. I guess it doesn't matter too much, but I'd think that there should be one canonical source. Move this back to `add-remotes`?